contracts/rust/src/lib.rs

```diff
@@ -104,6 +111,23 @@ impl NEP4 for NonFungibleTokenBasic {
 
     fn transfer(&mut self, new_owner_id: AccountId, token_id: TokenId) {
         let token_owner_account_id = self.get_token_owner(token_id);
+        let signer = env::predecessor_account_id();
+        if signer != token_owner_account_id {
+            env::panic(b"Attempt to call transfer on tokens beloning to another account.")
+        }
+
+        if !self.check_access(token_owner_account_id) {
```

Comment:
I think that this second check is unnecessary? We've just determined that the predecessor and token owner are one and the same

Reply:
Removed. It looks like there was no test coverage for that line, so we also improved code coverage yay!